Rakefile

```diff
@@ -0,0 +1,157 @@
+# Copyright 2015, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "git"
+require "octokit"
+require "netrc"
+require "yaml"
+
+
+task :configure do
+  config = YAML.load_file(File.join(File.dirname(__FILE__), "config.yml"))
+  @all_barclamps = config["barclamps"]["crowbar"] + config["barclamps"]["openstack"]
+end
+
+desc "This task setup the barclamps for development."
+task :init => [:fork, :clone, :add_upstream, :add_susecloud] do
+  # nothing to do here
+end
+
+desc "Update all clones."
+task :up => [:pull_upstream] do
+  # nothing to do here
+end
+
+desc "Forks all required barclamps to your home project."
+task :fork => [:configure] do
+  client = Octokit::Client.new(:netrc => true)
+
+  @all_barclamps.each do |barclamp|
+    if client.branches("#{client.login}/barclamp-#{barclamp}")
+      puts "Barclamp #{barclamp} already forked. Skipping..."
+    else
+      puts "Forking barclamp #{barclamp}..."
+      client.fork("crowbar/#{barclamp}")
+    end
+  end
+end
+
+desc "Clones all required barclamps into the barclamps folder."
+task :clone => [:configure] do
+  client = Octokit::Client.new(:netrc => true)
+
+  Dir.chdir("barclamps") do
+    @all_barclamps.each do |barclamp|
+      if Dir.exist?(barclamp)
+        puts "Barclamp #{barclamp} already exists. Skipping..."
+      else
+        puts "Cloning barclamp #{barclamp}..."
+        Git.clone("git@github.com:#{client.login}/barclamp-#{barclamp}.git", barclamp)
+      end
+    end
+  end
+end
+
+desc "Add upstream remote to the crowbar organisation."
+task :add_upstream => [:configure] do
+  Dir.chdir("barclamps") do
+    @all_barclamps.each do |barclamp|
+      Dir.chdir(barclamp) do
+        git = Git::Lib.new()
```

Comment:
yes, it is a direct instanciation of the Lib Object which provides all necessary methods